course02/homework01/linked_list.h

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+
+using namespace std;
+
+class LinkedList
+{
+
+public:
+	LinkedList();
+	~LinkedList();
+
+	struct Node
+	{
+		Node* next{ nullptr };
+		int data;
+	};
+
+	void Insert(int key);
+	void Remove(int key);
+	Node* Find(int key);
+	const Node* End() const;
+	std::size_t Size() const;
+	bool Empty() const;
+
+	friend ostream &operator<<(ostream&, LinkedList&);
+
+private:
+
+	void clear();
+	Node* find_previous(int k);
+
+	Node* mHead{ nullptr };
+	Node* mTail{ nullptr };
+	std::size_t mSize = 0;
+};
+
+inline ostream &operator<<(ostream &output, LinkedList& linkedList) {
+	
+	LinkedList::Node* node = new LinkedList::Node;
```

Comment:
A stream operator that constructors new node is rather unacceptable! Why not just iterating through the nodes?

Reply:
I see that this is a bug; you thus leak memory on every call. That's your second bug caused by raw pointers, congratulations :)